src/input/mod.rs

```diff
@@ -213,6 +213,25 @@ impl State {
                     self.common.idle_notifier_state.notify_activity(&seat);
 
                     let keycode = event.key_code();
+
+                    const POWER_OFF: u32 = 116;
+                    // if power key is pressed, just use the system action for power off
+                    // this is a workaround for the fact that the power key is not
+                    // available in the keymap, so we cannot use the keymap to determine
+                    // if the key is pressed
+                    if keycode.raw() == POWER_OFF + X11_KEYCODE_OFFSET {
```

Comment:
We probably also want to check `event.state()` to be `Pressed` to not trigger this twice.